web/tests/functional/authentication/test_permission_management.py

```diff
@@ -0,0 +1,220 @@
+
+        # Display test name to log.
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self._test_workspace)
+
+        # Get the test configuration from the prepared int the test workspace.
+        test_config = env.import_test_cfg(self._test_workspace)
+
+        # Log in and create credential file (if it is not exists).
+        superuser_name = "root"
+        superuser_passwd = "root"
+
+        codechecker.login(test_config['codechecker_cfg'],
+                          self._test_workspace, superuser_name,
+                          superuser_passwd)
+
+        temp_auth_client = env.setup_auth_client(self._test_workspace,
+                                                 session_token='_PROHIBIT')
+
+        # A non-authenticated session should return an empty user.
+        user_name = temp_auth_client.getLoggedInUser()
+        self.assertEqual(user_name, "")
+
+        # Create a SUPERUSER login.
+        self._session_token = temp_auth_client.performLogin(
+            "Username:Password", superuser_name + ":" + superuser_passwd)
+        self.assertIsNotNone(self._session_token,
+                             "Valid credentials didn't give us a token!")
+
+        # Connect as root to the server.
+        self._root_auth_client = env.setup_auth_client(
+            self._test_workspace, session_token=self._session_token)
+        user_name = self._root_auth_client.getLoggedInUser()
+        self.assertEqual(user_name, superuser_name)
+
+        # Query data of current product
+        self._product_name = test_config['codechecker_cfg']['viewer_product']
+        self._product_client = env.setup_product_client(
+            self._test_workspace, product=self._product_name)
+        self.assertIsNotNone(self._product_client)
+        self._current_product = self._product_client.getCurrentProduct()
+        self.assertIsNotNone(self._current_product)
+        self._product_id = self._current_product.id
+        self.assertIsNotNone(self._product_id)
+
+        extra_params = {'productID': self._product_id}
+        self._extra_params_as_json = json.dumps(extra_params)
+
+    def test_product_permissions(self):
+        product_permissions = self._root_auth_client.getPermissions('PRODUCT')
+        product_permission_set = set(product_permissions)
+        expected_permission_set = set(
+            [Permission.PRODUCT_ADMIN, Permission.PRODUCT_ACCESS,
+             Permission.PRODUCT_STORE])
+        self.assertSetEqual(
+            expected_permission_set, product_permission_set,
+            "PRODUCT permission set in the database is different.")
+
+    def test_system_permissions(self):
+        sys_permissions = self._root_auth_client.getPermissions('SYSTEM')
+        sys_permission_set = set(sys_permissions)
+        expected_permission_set = set(
+            [Permission.SUPERUSER])
+        self.assertSetEqual(
+            expected_permission_set, sys_permission_set,
```

Comment:
Something similar would be better instead of creating separate variables:
```suggestion
            set(expected_permissions), set(sys_permissions),
```
Or you can initialize variables with set: `sys_permissions = set(self._root_auth_client.getPermissions('SYSTEM'))`

Reply:
I did not eliminate variable names, because they help understand the intent for the reader.